stdlib/tkinter/__init__.pyi

```diff
@@ -1112,17 +1115,137 @@ class Canvas(Widget, XView, YView):
+        disableddash: _Color = ...,
+        disabledfill: _Color = ...,
+        disabledoutline: _Color = ...,
+        disabledoutlinestipple: _Color = ...,
+        disabledstipple: _Bitmap = ...,
+        disabledwidth: Union[int, str]  = ...,
+        fill: _Color = ...,
+        joinstyle: Literal["round", "bevel", "miter"] = ...,
+        offset: Union[int, str]  = ...,
+        outline: _Color = ...,
+        outlineoffset: Union[int, str]  = ...,
+        outlinestipple: _Bitmap = ...,
+        smooth: Literal[0, 1] = ...,
+        splinesteps: int = ...,
+        state: Literal["normal", "active", "disabled"] = ...,
+        stipple: _Bitmap = ...,
+        tags: Union[str, Tuple[str, ...]] = ...,
+        width: Union[int, str]  = ...,
+    ) -> _ObjectId: ...
+    def create_rectangle(self, x0: int, y0: int, x1: int, y1: int, /,
+        activedash: _Color = ...,
+        activefill: _Color = ...,
+        activeoutline: _Color = ...,
+        activeoutlinestipple: _Color = ...,
+        activestipple: str = ...,
+        activewidth: Union[int, str] = ...,
+        dash: Union[Tuple[int], Tuple[int, int], Tuple[int, int, int, int]] = ...,
+        dashoffset: Union[int, str] = ...,
+        disableddash: _Color = ...,
+        disabledfill: _Color = ...,
+        disabledoutline: _Color = ...,
+        disabledoutlinestipple: _Color = ...,
+        disabledstipple: _Bitmap = ...,
+        disabledwidth: Union[int, str]  = ...,
+        fill: _Color = ...,
+        offset: Union[int, str]  = ...,
+        outline: _Color = ...,
+        outlineoffset: Union[int, str]  = ...,
+        outlinestipple: _Bitmap = ...,
+        state: Literal["normal", "active", "disabled"] = ...,
+        stipple: _Bitmap = ...,
+        tags: Union[str, Tuple[str, ...]] = ...,
+        width: Union[int, str]  = ...,
+    ) -> _ObjectId: ...
+    def create_text(self, x: int, y: int, /
+        activefill: _Color = ...,
+        activestipple: str = ...,
+        anchor: _Anchor = ...,
+        disabledfill: _Color = ...,
+        disabledstipple: _Bitmap = ...,
+        fill: _Color = ...,
+        font: _FontDescription = ...,
+        justify: Literal["left", "center", "right"] = ...,
+        offset: Union[int, str]  = ...,
+        state: Literal["normal", "active", "disabled"] = ...,
+        stipple: _Bitmap = ...,
+        tags: Union[str, Tuple[str, ...]] = ...,
+        text: str = ...,
+        width: Union[int, str]  = ...,
+    ) -> _ObjectId: ...
+    def create_window(self, x: int, y: int, /,
+        anchor: _Anchor = ...,
+        height: Union[int, str] = ...,
+        state: Literal["normal", "active", "disabled"] = ...,
+        tags: Union[str, Tuple[str, ...]] = ...,
+        width: Union[int, str]  = ...,
+        window: Widget = ...,
+    ) -> _ObjectId: ...
     def dchars(self, *args): ...
-    def delete(self, *args): ...
+    def delete(self, *args: Union[_ObjectId, typing.Iterable[_ObjectId]]) -> None: ...
```

Comment:
Nit: Import `Iterable` from `collections.abc` and just use `Iterable` here.

Reply:
Please use `_TkinterSequence`

Reply:
Hm. Should `_TkinterSequence` be redefined as this?

```diff
- _TkinterSequence = Union[List[_T], Tuple[_T, ...]]
+ _TkinterSequence = Union[List[_T], Iterable[_T, ...]]
```

Reply:
No. Tkinter doesn't accept all iterables. There's a comment showing how it doesn't accept `collections.deque`.

Reply:
Okay, it's just that `delete` definitely seems to operate on any iterable recursively.

Reply:
The source code is:

```
    def delete(self, *args):
        """Delete items identified by all tag or ids contained in ARGS."""
        self.tk.call((self._w, 'delete') + args)
```

So each item of `args` is passed to `self.tk.call`, which wants `_TkinterSequence` (but silently does the wrong thing with e.g. deques):

```
>>> import tkinter, collections
>>> c = tkinter.Canvas()
>>> c.pack()
>>> c.create_rectangle(100, 100, 200, 200)
1
>>> c.delete([1])   # rectangle goes away
>>> c.create_rectangle(100, 100, 200, 200)
2
>>> c.delete(collections.deque([2]))  # rectangle DOES NOT GO AWAY
```

Reply:
Actually, `c.delete([3, 4])` doesn't work either, so we shouldn't use `_TkinterSequence` or `Iterable` at all.